scenario/mocking.py

```diff
@@ -424,6 +427,9 @@ def secret_info_get(
         label: Optional[str] = None,
     ) -> SecretInfo:
         secret = self._get_secret(id, label)
+        # If both the id and label are provided, then update the label.
```

Comment:
should we be doing this in `_get_secret`?
we-re side-effecting anyway, so...

Reply:
I wondered about that. `_get_secret` is used by three other methods - for grant/revoke it's not relevant, but the code would skipped anyway because the label will always be None. For set, it is relevant, but it's already done because it's updating potentially all the metadata.

So it's either updated twice in set or there's the code duplication. Maybe it's just cleaner in `_get_secret` so that's the better choice.

Reply:
mmm right. I'm 50-50 on this one, I don't mind either way ;)